File: ios/epocheye/AR/ARKitModule.m
#import <React/RCTBridgeModule.h>

@interface RCT_EXTERN_MODULE(ARKitModule, NSObject)

RCT_EXTERN_METHOD(isAvailable:(RCTPromiseResolveBlock)resolve
                  rejecter:(RCTPromiseRejectBlock)reject)

RCT_EXTERN_METHOD(isInstalled:(RCTPromiseResolveBlock)resolve
                  rejecter:(RCTPromiseRejectBlock)reject)

@end

File: ios/epocheye/AR/ARKitModule.swift
import ARKit
import Foundation

@objc(ARKitModule)
final class ARKitModule: NSObject {

  @objc static func requiresMainQueueSetup() -> Bool {
    return true
  }

  @objc(isAvailable:rejecter:)
  func isAvailable(_ resolve: @escaping RCTPromiseResolveBlock,
                   rejecter reject: @escaping RCTPromiseRejectBlock) {
    resolve(ARWorldTrackingConfiguration.isSupported)
  }

  @objc(isInstalled:rejecter:)
  func isInstalled(_ resolve: @escaping RCTPromiseResolveBlock,
                   rejecter reject: @escaping RCTPromiseRejectBlock) {
    // ARKit ships with iOS; availability == installed.
    resolve(ARWorldTrackingConfiguration.isSupported)
  }
}
